### DISCUSSION TITLE
Classes in different order on server vs client

### DISCUSSION THREAD

**Discussion by mattstobbs** [vercel/next.js#35261] [Help]
I have a component that I'm using to render CodeBlocks. It looks like this:

```jsx
import { highlightCode } from './utils';

export default function CodeBlock({ codeString, language }) {
  return (
    <pre className={`language-${language} other-classes`}>
      <code
        className={`language-${language}`}
        dangerouslySetInnerHTML={{
          __html: highlightCode(codeString, language),
        }}
      />
    </pre>
  );
}
```

However, when I run the code with `npm run dev`, I get an error in the console:
> Warning: Prop `className` did not match. Server: "other-classes language-tsx" Client: "language-tsx other-classes"

Any ideas on what could be causing this? I appreciate I haven't given enough details, but I have no clue where to even start looking for the source of the issue. Why would the ordering of the classes within the string be different on the server?

**Comment by icyJoseph** [comment]
At first glance, the error can't be here. How's this component being used? 

What happens if you do:

```jsx
export default function CodeBlock({ codeString, language }) {
 const preClassName = `language-${language} other-classes`;
  return (
    <pre className={preClassName}>
      <code
        className={`language-${language}`}
        dangerouslySetInnerHTML={{
          __html: highlightCode(codeString, language),
        }}
      />
    </pre>
  );
}
```

Perhaps even log that `preClassName`. What Next.js version are you using? And could we look at the `next.config.js` file?

**Reply by mattstobbs** [reply]
Thanks for your answer. I think the problem is with `highlightCode` as when I replace it with just a string I don't see the error. It may be something to do with Prism but I have no idea. I've started using https://github.com/FormidableLabs/prism-react-renderer now though and I'm no longer seeing the problem so it doesn't matter now. Thanks for your help though!
